Add comments to handler for async handshake (#1019)
cosmos/ics: spec/core/ics-026-routing-module/README.md

    datagram.portIdentifier,
    datagram.counterpartyPortIdentifier
  )

>   // SYNCHRONOUS: the following calls happen syncrhonously with the call above

Reviewer: Minor typo here and in the below `SYNCHRONOUS:` comments.
```suggestion
  // SYNCHRONOUS: the following calls happen synchronously with the call above
```

  // ASYNCHRONOUS: the module callback will be called at a time later than the channel handler

>   // in this case, the channel identifier will be stored with a sentinel value so it is not taken

Reviewer 2: With this you mean that a sentinel value is stored in the path `channelEnds/ports/{datagram.portIdentifier}/channels/{channelIdentifier}`, right? Maybe it's more explicit to mention it like this?

```

```typescript
function handleChanCloseConfirm(datagram: ChanCloseConfirm) {  
    handler.chanCloseConfirm(
      datagram.portIdentifier,
      datagram.channelIdentifier,
      datagram.proofInit,
      datagram.proofHeight
    )
    // SYNCHRONOUS: the following calls happen syncrhonously with the call above
    // ASYNCHRONOUS: the module callback will be called at a time later than the channel handler
    // in this case, the channel identifier will be stored with a sentinel value so it is not taken

>     // by a new channel handshake and the capability is reserved for the application module.

Reviewer 2: Is this actually needed?

```

```typescript
function handleChanCloseConfirm(datagram: ChanCloseConfirm) {  
    handler.chanCloseConfirm(
      datagram.portIdentifier,
      datagram.channelIdentifier,
      datagram.proofInit,
      datagram.proofHeight
    )
    // SYNCHRONOUS: the following calls happen syncrhonously with the call above
    // ASYNCHRONOUS: the module callback will be called at a time later than the channel handler
    // in this case, the channel identifier will be stored with a sentinel value so it is not taken
    // by a new channel handshake and the capability is reserved for the application module.
    // When the module eventually executes its callback it must call writeChannel so that the channel

>     // can be written into an OPEN state and the handshake can proceed on the counterparty.

Reviewer: ```suggestion
    // can be written into a CLOSED state and the handshake can proceed on the counterparty.
```
